File: frontend/demo/component/icons/vaadin-icons.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';
import '@vaadin/icon';
import '@vaadin/icons';
import '@vaadin/button';
import '@vaadin/horizontal-layout';

@customElement('vaadin-icons')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-horizontal-layout theme="spacing" class="items-center">
        <!-- tag::snippet[] -->
        <vaadin-icon icon="vaadin:phone"></vaadin-icon>
        <vaadin-icon icon="vaadin:calendar"></vaadin-icon>
        <vaadin-icon icon="vaadin:alarm"></vaadin-icon>
        <vaadin-button>
          <vaadin-icon icon="vaadin:bell"></vaadin-icon>
        </vaadin-button>
        <!-- end::snippet[] -->
      </vaadin-horizontal-layout>
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/icons/VaadinIcons.java
package com.vaadin.demo.component.icons;

import com.vaadin.flow.router.Route;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.icon.Icon;
import com.vaadin.flow.component.icon.VaadinIcon;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("vaadin-icons")
public class VaadinIcons extends Div {

  public VaadinIcons() {
    HorizontalLayout layout = new HorizontalLayout();
    layout.setSpacing(true);
    layout.addClassName("items-center");

    // tag::snippet[]
    Icon phoneIcon = VaadinIcon.PHONE.create();
    Icon calendarIcon = VaadinIcon.CALENDAR.create();
    Icon alarmIcon = VaadinIcon.ALARM.create();
    Button button = new Button(VaadinIcon.BELL.create());

    layout.add(phoneIcon, calendarIcon, alarmIcon, button);
    // end::snippet[]

    add(layout);
  }

  public static class Exporter extends DemoExporter<VaadinIcons> { // hidden-source-line
  }
}
